internal/controller/location.go

```diff
@@ -0,0 +1,30 @@
+package controller
+
+import (
+	"go-rest-template/internal/domain"
+	"go-rest-template/internal/service"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+type LocationController interface {
+	Create(ctx *gin.Context)
+}
+type locationController struct {
+	svc service.LocationService
+}
+
+func NewLocationController(svc service.LocationService) LocationController {
+	return &locationController{svc: svc}
+}
+
+func (c *locationController) Create(ctx *gin.Context) {
+	var p domain.Payload
+	if !p.IsValid() {
+		ctx.String(http.StatusBadRequest, "Data incorrecta")
```

Comment:
Siempre se devuelve un JSON, si devolvés un string a veces y un JSON otras se le rompe al que te está usando la API porque es inconsistente el tipo.
```suggestion
		ctx.JSON(http.StatusBadRequest, gin.H{"message":"invalid payload"})
```

Reply:
Hecho [3a28aad](https://github.com/luxarts/hte-location-ms/pull/1/commits/3a28aadcc7a306c09dfe52eca6df81031c9b330b)